From: hbabu@us.ibm.com
Subject: Bug 157018 - LTC21952-kdump boot is not successful on JS21

without it, the kdump kernel hangs very very early, before printing the Linux version
banner.
with it, kdump boot via sysrq c is successful.
but no keyboard input accepted (this needs a different patch)

Signed-off-by: Olaf Hering <olh@suse.de>
---
 arch/powerpc/kernel/iommu.c           |    2 ++
 arch/powerpc/platforms/pseries/xics.c |    1 +
 2 files changed, 3 insertions(+)

Index: linux-2.6.16-olh/arch/powerpc/kernel/iommu.c
===================================================================
--- linux-2.6.16-olh.orig/arch/powerpc/kernel/iommu.c
+++ linux-2.6.16-olh/arch/powerpc/kernel/iommu.c
@@ -427,8 +427,10 @@ struct iommu_table *iommu_init_table(str
 	tbl->it_largehint = tbl->it_halfpoint;
 	spin_lock_init(&tbl->it_lock);
 
+#ifndef CONFIG_CRASH_DUMP
 	/* Clear the hardware table in case firmware left allocations in it */
 	ppc_md.tce_free(tbl, tbl->it_offset, tbl->it_size);
+#endif
 
 	if (!welcomed) {
 		printk(KERN_INFO "IOMMU table initialized, virtual merging %s\n",
Index: linux-2.6.16-olh/arch/powerpc/platforms/pseries/xics.c
===================================================================
--- linux-2.6.16-olh.orig/arch/powerpc/platforms/pseries/xics.c
+++ linux-2.6.16-olh/arch/powerpc/platforms/pseries/xics.c
@@ -640,6 +640,7 @@ void xics_teardown_cpu(int secondary)
 	ops->cppr_info(cpu, 0x00);
 	iosync();
 
+	ops->qirr_info(cpu, 0xff);
 	/*
 	 * Some machines need to have at least one cpu in the GIQ,
 	 * so leave the master cpu in the group.
